nixos/modules/services/misc/sourcehut/default.nix

```diff
@@ -0,0 +1,173 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.sourcehut;
+  cfgIni = cfg.settings;
+
+  # Specialized python containing all the modules
+  python = pkgs.sourcehut.python.withPackages (ps: with ps; [
+    gunicorn
+    # Sourcehut services
+    buildsrht dispatchsrht gitsrht hgsrht hubsrht listssrht mansrht
+    metasrht pastesrht todosrht
+  ]);
+in {
+  imports =
+    [
+      ./git.nix
+      ./hg.nix
+      ./hub.nix
+      ./todo.nix
+      ./man.nix
+      ./meta.nix
+      ./paste.nix
+      ./builds.nix
+      ./lists.nix
+    ];
+
+  options.services.sourcehut = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Enable sourcehut - git hosting, continuous integration, mailing list, ticket tracking,
+        task dispatching, wiki and account management services.
+      '';
+    };
+
+    services = mkOption {
+      type = types.nonEmptyListOf (types.enum [ "builds" "dispatch" "git" "hub" "hg" "lists" "man" "meta" "paste" "todo" ]);
+      default = [ "builds" "dispatch" "git" "hub" "hg" "lists" "man" "meta" "paste" "todo" ];
```

Comment:
Maybe a little smaller default?
I don't see a reason why you would want to run git and hg.
How about [ "builds" "dispatch" "git" ]
Just a thought

Reply:
hrm.. i don't have dispatch working yet.... adding it to the TODO